keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -0,0 +1,225 @@
+- Multiple developers have blogged how to use this feature, but all are on standalone machines instead of orchestrated through Kubernetes
+  - https://artisticcheese.wordpress.com/2017/09/09/enabling-integrated-windows-authentication-in-windows-docker-container/
+  - https://community.cireson.com/discussion/3853/example-cireson-scsm-portal-on-docker-windows-containers
+  - https://cloudiqtech.com/windows-2016-docker-containers-using-gmsa-connect-to-sql-server/
+  - https://success.docker.com/api/asset/.%2Fmodernizing-traditional-dot-net-applications%2F%23integratedwindowsauthentication 
+
+
+### Goals
+
+- Windows users can run containers with an existing GMSA identity on Kubernetes
+- No extra files or Windows registry keys are needed on each Windows node. All needed data should flow through Kubernetes+Kubelet APIs
+- Provide a reasonable effort to prevent pods from being inadvertently scheduled with access to a GMSA that the operator shouldn't have rights to
+
+### Non-Goals
+
+- Running Linux applications using GMSA or a general Kerberos based authentication system
+- Replacing any existing Kubernetes authorization or authentication controls
+- Providing unique container identities. By design, Windows GMSA are used where multiple nodes are running apps as the same Active Directory principal.
+- Isolation between container users and processes running as the GMSA. Windows already allows users and system services with sufficient privilege to create processes using a GMSA.
+- Preventing the node from having access to the GMSA. Since the node already has authorization to access the GMSA, it can start processes or services using as the GMSA. Containers do not change this behavior.
+
+
+## Proposal
+
+### Background
+
+#### What is Active Directory?
+Windows applications and services typically use Active Directory identities to facilitate authentication and authorization between resources. In a traditional virtual machine scenario, the computer is joined to an Active Directory domain which enables it to use Kerberos, NTLM, and LDAP to identify and query for information about other resources in the domain. When a computer is joined to Active Directory, it is given an unique identity represented as a computer object in LDAP.
+
+#### What is a Windows service account?
+A Group Managed Service Account (GMSA) is a shared Active Directory identity that enables common scenarios such as authenticating and authorizing incoming requests and accessing downstream resources such as a database server, file share, or other workload. It can be used by multiple authorized users or computers at the same time.
+
+#### How is it applied to containers?
+To achieve the scale and speed required for containers, Windows uses a group managed service account in lieu of individual computer accounts to enable Windows containers to communicate with Active Directory. As of right now, Windows cannot use individual Active Directory computer & user accounts.
```

Comment:
> As of right now, Windows cannot use individual Active Directory computer & user accounts.

What does this mean?

Reply:
Active Directory supports various types of accounts like: computer accounts, user accounts and service accounts. We can clarify the above statement to indicate that the container subsystem on Windows (Host Compute Service) only supports GMSA as a credential spec for containers and add a link to https://github.com/opencontainers/runtime-spec/blob/master/config-windows.md#credential-spec